### PR TITLE
Pay 5504 template selection based on instruction type

### PR DESCRIPTION
**Pull request by sayali-salunkhe**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/refunds/services/RefundsServiceImpl.java [hunk 1]
```diff
@@ -117,9 +117,20 @@
 
     @Override
     public RefundResponse initiateRefund(RefundRequest refundRequest, MultiValueMap<String, String> headers) throws CheckDigitException {
-        //validateRefundRequest(refundRequest); //disabled this validation to allow partial refunds
+
+        String paymnetMethod = null;
```
Comment (kaushikmaitra) [inline src/main/java/uk/gov/hmcts/reform/refunds/services/RefundsServiceImpl.java]: Pls correct the spelling

File: src/main/java/uk/gov/hmcts/reform/refunds/services/RefundsServiceImpl.java [hunk 2]
```diff
@@ -117,9 +117,20 @@
...
+
+        if (refundRequest.getPaymentMethod() != null) {
+
+            if (refundRequest.getPaymentMethod().equals("cheque") || refundRequest.getPaymentMethod().equals("cash")
+                || refundRequest.getPaymentMethod().equals("postal order") && (refundRequest.getPaymentChannel().equals("bulk scan"))) {
```
Comment (kaushikmaitra) [inline src/main/java/uk/gov/hmcts/reform/refunds/services/RefundsServiceImpl.java]: Pls remove hard-coding and use constants

File: src/main/java/uk/gov/hmcts/reform/refunds/services/RefundsServiceImpl.java [hunk 3]
```diff
@@ -117,9 +117,20 @@
 
     @Override
     public RefundResponse initiateRefund(RefundRequest refundRequest, MultiValueMap<String, String> headers) throws CheckDigitException {
-        //validateRefundRequest(refundRequest); //disabled this validation to allow partial refunds
+
+        String paymnetMethod = null;
+
+        if (refundRequest.getPaymentMethod() != null) {
+
+            if (refundRequest.getPaymentMethod().equals("cheque") || refundRequest.getPaymentMethod().equals("cash")
```
[the remark this answers is not in the document]
Comment (sayali-salunkhe) [reply src/main/java/uk/gov/hmcts/reform/refunds/services/RefundsServiceImpl.java]: This code is updated in base branch - **_instruction_type_** , hence issue resolved automatically

File: src/main/java/uk/gov/hmcts/reform/refunds/services/RefundsServiceImpl.java (hunk 2 above)
Comment (sayali-salunkhe) [reply src/main/java/uk/gov/hmcts/reform/refunds/services/RefundsServiceImpl.java]: This code is updated in base branch - **_instruction_type_** , hence issue resolved automatically

File: src/main/java/uk/gov/hmcts/reform/refunds/utils/RefundsUtil.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    private String cardPbaLetterTemplateId;
+
+    @Value("${notify.template.card-pba.email}")
+    private String cardPbaEmailTemplateId;
+
+    public String getTemplate(Refund refund) {
+        String templateId = null;
+        if (null != refund.getRefundInstructionType()) {
+
+            if ("RefundWhenContacted".equals(refund.getRefundInstructionType())) {
```
[the remark this answers is not in the document]
Comment (sayali-salunkhe) [reply src/main/java/uk/gov/hmcts/reform/refunds/utils/RefundsUtil.java]: Updated code

**Review by kaushikmaitra — APPROVED** [review]

**Pull request merged**